cert-tool/cert-tool.sh

```diff
@@ -449,6 +457,61 @@ function load_certs() {
+# ARGS: none
+# OUTS: None if successful, Error text otherwise
+function load_current_state() {
+    INGRESS_GENERATION=$(${oc_cmd} get deployment router-default -n openshift-ingress -o json | jq -r '.status.observedGeneration')
+    INGRESS_REPLICAS=$(${oc_cmd} get deployment router-default -n openshift-ingress -o json | jq -r '.status.replicas')
+    INGRESS_AVAILABLEREPLICAS=$(${oc_cmd} get deployment router-default -n openshift-ingress -o json | jq -r '.status.availableReplicas')
+    INGRESS_READYREPLICAS=$(${oc_cmd} get deployment router-default -n openshift-ingress -o json | jq -r '.status.readyReplicas')
+    INGRESS_UPDATEDREPLICAS=$(${oc_cmd} get deployment router-default -n openshift-ingress -o json | jq -r '.status.updatedReplicas')
+
+    NEXT_INGRESS_GENERATION=$((INGRESS_GENERATION + 1))
+
+    read -r API_NAME API_VERSION API_AVAILABLE API_PROGRESSING API_DEGRADED API_SINCE <<< "$(${oc_cmd} get clusteroperators kube-apiserver | grep -v 'NAME')"
+}
+
+
+# DESC: Load current state
+# ARGS: none
+# OUTS: None if successful, Error text otherwise
+function validate_state() {
+    SLEEP=60
+    INGRESS_COMPLETE=false
+    if [[ ${INGRESS_PATCHED} == false ]]; then
+        INGRESS_COMPLETE=true
+    fi
+    API_COMPLETE=false
+    if [[ ${API_PATCHED} == false ]]; then
+        API_COMPLETE=true
+    fi
+    for ITER in {0..15}
+    do
+        if [[ ${INGRESS_COMPLETE} == true && ${API_COMPLETE} == true ]]; then
+            break
+        fi
+
+        # Checking default ingress status
+        C_INGRESS_GENERATION=$(${oc_cmd} get deployment router-default -n openshift-ingress -o json | jq -r '.status.observedGeneration')
+        C_INGRESS_REPLICAS=$(${oc_cmd} get deployment router-default -n openshift-ingress -o json | jq -r '.status.replicas')
+        C_INGRESS_AVAILABLEREPLICAS=$(${oc_cmd} get deployment router-default -n openshift-ingress -o json | jq -r '.status.availableReplicas')
+        C_INGRESS_READYREPLICAS=$(${oc_cmd} get deployment router-default -n openshift-ingress -o json | jq -r '.status.readyReplicas')
+        C_INGRESS_UPDATEDREPLICAS=$(${oc_cmd} get deployment router-default -n openshift-ingress -o json | jq -r '.status.updatedReplicas')
+
+        if [[ ${C_INGRESS_GENERATION} -eq ${NEXT_INGRESS_GENERATION} && ${INGRESS_REPLICAS} -eq ${C_INGRESS_REPLICAS} && ${INGRESS_AVAILABLEREPLICAS} -eq ${C_INGRESS_AVAILABLEREPLICAS} && ${INGRESS_READYREPLICAS} -eq ${C_INGRESS_READYREPLICAS} && ${INGRESS_UPDATEDREPLICAS} -eq ${C_INGRESS_UPDATEDREPLICAS} ]]; then
+            INGRESS_COMPLETE=true
+        fi
+        # Checking kube-apiserver operator status
+        read -r C_API_NAME C_API_VERSION C_API_AVAILABLE C_API_PROGRESSING C_API_DEGRADED C_API_SINCE <<< "$(${oc_cmd} get clusteroperators kube-apiserver | grep -v 'NAME')"
+        if [[ ${C_API_AVAILABLE} == "True" && ${C_API_PROGRESSING} == "False" && ${C_API_DEGRADED} == "False" ]]; then
+            API_COMPLETE=true
+        fi
+
+        sleep ${SLEEP}
```

Comment:
Is there a need for this sleep? From what I can see, this function is called only once.

Reply:
There is a for loop - which will keep asking for status every 60 seconds (total 16 iterations)
Mostly for kubeapi-server to complete transition